common/config.go

```diff
@@ -705,9 +705,33 @@ func (c *EvmUpstreamConfig) Copy() *EvmUpstreamConfig {
 	return copied
 }
 
+type MatcherConfig struct {
+	Network  string                 `yaml:"network,omitempty" json:"network"`
+	Method   string                 `yaml:"method,omitempty" json:"method"`
+	Params   []interface{}          `yaml:"params,omitempty" json:"params"`
+	Finality DataFinalityStateSlice `yaml:"finality,omitempty" json:"finality" tstype:"DataFinalityState[]"`
```

Comment:
```suggestion
	Finality []DataFinalityState `yaml:"finality,omitempty" json:"finality" tstype:"DataFinalityState[]"`
```
Any downsides to something like this?